From 0a2093dcc5e65d37b21e44fd90c6c17da78d09ec Mon Sep 17 00:00:00 2001
From: dzchoi <19231860+dzchoi@users.noreply.github.com>
Date: Sun, 9 Mar 2025 13:02:08 -0400
Subject: [PATCH 7/8] [PATCH 7/7] Default to 32 bit build and small buffer
 size.

Set the word length and buffer size in luaconf.h so that it doesn't have to
be specified in the compiler's command line each time.

LUAL_BUFFERSIZE can still be overrriden by -DLUAL_BUFFERSIZE=xxxx.
To force a 64 bit build, define LUA_64BIT.
---
 luaconf.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/luaconf.h b/luaconf.h
index 3ff64b7..8293893 100644
--- a/luaconf.h
+++ b/luaconf.h
@@ -33,7 +33,9 @@
 ** ensure that all software connected to Lua will be compiled with the
 ** same configuration.
 */
-/* #define LUA_32BITS */
+#ifndef LUA_64BITS
+#define LUA_32BITS
+#endif
 
 
 /*
@@ -760,7 +762,7 @@
 #if LUA_FLOAT_TYPE == LUA_FLOAT_LONGDOUBLE
 #define LUAL_BUFFERSIZE		8192
 #else
-#define LUAL_BUFFERSIZE   ((int)(0x80 * sizeof(void*) * sizeof(lua_Integer)))
+#define LUAL_BUFFERSIZE   64
 #endif
 
 #endif
-- 
2.34.1

